docs/enhancements/support-node-labeling.md

```diff
@@ -0,0 +1,128 @@
+---
+title: node labeling support
+authors:
+- "@oamizur"
+  creation-date: 2022-12-20
+  last-updated: 2022-12-20
+---
+
+# Support node labeling during installation
+
+## Summary
+
+There was a requirement to support node roles during installation.  These roles 
```

Comment:
roles is just a use case here, i suggest to have summary talk about the general use case, which is defining labels on the agent that will result in the nodes on the spoke cluster after the installation

Reply:
You can't set custom role before node joins

Reply:
These roles do not have operational meaning from our POV.  These will be user defined attributes for their own use.

Reply:
This is depends on what you mean by roles. I talked about k8s role and you can set them only after node joins k8s cluster